### ISSUE TITLE
Does the SZ3 CLI support shared-memory parallellism?

### ISSUE THREAD

**Issue by Wout12345** [szcompressor/SZ3#8]
Does SZ3 support shared-memory parallellism (from the command-line interface)? For example, using OpenMP, since I saw it's used a bit in the code, however from the README it's not clear if multi-threading is enabled by default (if OpenMP is found), and if so, if we can control the number of threads. Are there any other restrictions? From my (short) observations in htop, it seems like SZ only uses 1 CPU by default. It would be very nice if we could get more information about this.

**Comment by ayzk** [comment]
@Wout12345 Yes, the building process will automatically compile the OpenMP code. To use OpenMP, please enabled it in the configuration file (OpenMP = Yes) and use the -c option to pass the configuration file to sz. The number of threads can be controlled by the OpenMP environment variable omp_num_threads.

**Comment by Wout12345** [comment]
Thank you for your response, but unfortunately this doesn't seem to work for me. I created the following configuration file at `../compressors/sz-config.txt`:
```
OpenMP = Yes
```
And am now calling SZ compression with the following command:
```
export OMP_NUM_THREADS=3
../compressors/sz -i /local_scratch/vsc32681-compression-comparison/data/Moffett-Field -z /local_scratch/vsc32681-compression-comparison/compressed-verify-setup -f -3 224 1024 512 -M PSNR -S 104.05666947148904 -c ../compressors/sz-config.txt
```
But htop shows that the SZ process always uses at most 100% CPU. Furthermore, when I run the command with time, I get output like:
```
Load cfg from ../compressors/sz-config.txt
====================================== BEGIN TUNING ================================
interp best interpAlgo = CUBIC
interp best direction = 0
choose Lorenzo
====================================== END TUNING ======================================
compression ratio = 4.68 
compression time = 4.879535
compressed data file = /local_scratch/vsc32681-compression-comparison/compressed-verify-setup

real	0m5.501s
user	0m4.270s
sys	0m1.035s
```
which shows no real multi-threading took place. I tried the same with SZ decompression but got similar results. I also tried replacing `Yes` by `YES` in my config file, to match the test configuration file better, however this also didn't solve anything.

Is this normal? Does SZ only use multi-threading for very large datasets? Or am I doing something else wrong?

**Comment by ayzk** [comment]
The configuration is in the INI format. Each parameter belongs to a section. This one should work.
[GlobalSettings]
OpenMP = YES

**Comment by Wout12345** [comment]
Thank you for the response, I wasn't aware of this. However, when I use this config file, I get the following output (to be sure, I also updated SZ to the latest version 3.1.4.0, so the output has a slightly different format now):
```
Error, error bound mode not supported
```
But when I set `OpenMP` to `NO`, I again get a normal output:
```
compression ratio = 4.68 
compression time = 5.362754
compressed data file = ../tmp/compressed-verify-setup
```
I also tried using the config file provided in the directory `test`, but I also get the same error when I both try to set `OpenMP` to `YES` and `ErrorBoundMode` to `PSNR`. Is there a way to combine parallellism with PSNR-based error control?
